src/components/ArtifactTimeline.vue

```diff
@@ -69,12 +101,23 @@
 
   const artifacts = ref<Artifact[]>([])
 
+  const getOffsetArtifacts = async (): Promise<void> => {
+    const result = await api.artifacts.getArtifacts(artifactsFilterWithOffset.value)
+    artifacts.value = [...new Map([...artifacts.value, ...result].map(obj => [obj.id, obj])).values()].sort((objA, objB) => sortDates(objB.created, objA.created))
+  }
+
   const getArtifacts = async (): Promise<void> => {
     const result = await api.artifacts.getArtifacts(artifactsFilter.value)
-    artifacts.value = [...artifacts.value, ...result]
+    artifacts.value = [...new Map([...artifacts.value, ...result].map(obj => [obj.id, obj])).values()].sort((objA, objB) => sortDates(objB.created, objA.created))
```

Comment:
1. You're repeating the same logic when setting artifacts.value
2. This feels very code golfy. 
3. Why create a map just to spread it into an array? Are you trying to remove duplicates?

Reply:
I can expand this a bit if it's helpful but yes the point is deduping since these are disparate calls - I'm not sure I understand your first point though

Reply:
Moved the sorting and mapping pieces out a little to make this a bit more readable